src/spyglass/spikesorting/spikesorting_curation.py

```diff
@@ -533,13 +546,54 @@ def _get_peak_channel(waveform_extractor: si.WaveformExtractor, peak_sign: str,
     peak_channel = {key : int(val) for key, val in peak_channel_dict.items()}
     return peak_channel
 
+def _get_cosine_similarity(waveform_extractor: si.WaveformExtractor, **metric_params):
+    """Gets the cosine similarity as computed by SpikeInterface's compute_template_similarity
+    """
+    cos_sim_mat = si.postprocessing.compute_template_similarity(
+        waveform_extractor=waveform_extractor,
+        **metric_params)
+    unit_ids = waveform_extractor.sorting.unit_ids
+    cos_sim_dict = {
```

Comment:
do we want to submit a full matrix or just the top 10, for example

Reply:
I'm not sure what people want/need, I was just starting with the broadest implementation.